app/src/main/java/org/mozilla/fenix/settings/logins/view/LoginDetailView.kt

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.logins.view
+
+import android.text.InputType
+import android.view.ViewGroup
+import androidx.core.content.res.ResourcesCompat
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.fragment_login_detail.*
+import kotlinx.android.synthetic.main.fragment_login_detail.view.*
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.logins.LoginsListState
+
+/**
+ * View that contains and configures the Login Details
+ */
+class LoginDetailView(override val containerView: ViewGroup) : LayoutContainer {
+    private val context = containerView.context
+
+    fun update(login: LoginsListState) {
+        webAddressText.text = login.currentItem?.origin
+        usernameText.text = login.currentItem?.username
+        passwordText.text = login.currentItem?.password
+    }
+
+    fun togglePasswordReveal(show: Boolean) {
+        if (show) showPassword() else { hidePassword() }
+    }
+
+    // TODO: create helper class for toggling passwords. Used in login detail and edit fragments.
+    fun showPassword() {
```

Comment:
I think this should be private.
This and the below should definitely not be publicly exposed to let others modify the View from outside.

But more than this, this shouldn't really exist in a MVI View.
The State received in the `update` method should be structured in such a way to allow the View to map it directly to it's properties. There is a 1-1 relation between them, it's ok for the State to be as granular and specialized as needed.
In this case, if we want such granularity for the passwordText the state should contain something like
- passwordTextToggleImage
- passwordTextContentDescription
- PasswordTextActualText

All of these being propagated through an Action from the Controller which should be the only one knowing about this logic - what should be displayed, when. 
Views should not be left to decide / choose what to display. They should render the received state.